src/Input.ts

```diff
@@ -0,0 +1,31 @@
+import * as fs from 'fs'
+
+import { Range } from './Range'
+
+export class Input {
+  public lines: string[]
+  constructor(public readonly path: string, public readonly text: string) {
+    this.lines = text.split('\n')
+  }
+
+  public static fromFile(path: string): Input {
+    // eslint-disable-next-line no-sync
+    return new Input(path, fs.readFileSync(path).toString())
+  }
+  public format(range: Range, diagnostic?: string): string {
```

Comment:
Short doc string? Bit hard to figure out what exactly this is doing based on the method body.